src/librustdoc/html/render.rs

```diff
@@ -359,6 +386,31 @@ impl Serialize for IndexItemFunctionType {
     }
 }
 
+#[derive(Debug)]
+pub struct TypeWithKind {
```

Comment:
Given this, why bother with `(Type, TypeKind)` in other places?

Reply:
Good point!

Reply:
I now have an answer: because they are `clean::Type` whereas we're using `render::Type`. This codebase needs some cleanup at this point... I'll rename the current `render::Type` to `render::RenderType` to remove confusion.